spyder/plugins/editor.py

```diff
@@ -1929,7 +1929,7 @@ def print_file(self):
                           header_font=self.get_plugin_font('printer_header'))
         printDialog = QPrintDialog(printer, editor)
         if editor.has_selected_text():
-            printDialog.addEnabledOption(QAbstractPrintDialog.PrintSelection)
+            printDialog.setOption(QAbstractPrintDialog.PrintSelection, True)

```

Comment:
From what I have read in the docs, `addEnabledOption` is and obsolet member of the class `QAbstractPrintDialog` (from QPrintDialog inherits), here the [link to the docs of Qt 4.8](https://doc.qt.io/qt-4.8/qabstractprintdialog-obsolete.html#addEnabledOption)